server/src/main/java/org/elasticsearch/search/crossproject/CrossProjectIndexResolutionValidator.java

```diff
@@ -103,82 +111,143 @@ public static ElasticsearchException validate(
-                }
                 assert localExpressions != ResolvedIndexExpression.LocalExpressions.NONE || false == remoteExpressions.isEmpty()
                     : "both local expression and remote expressions are empty which should have errored earlier at index rewriting time";
+                ElasticsearchSecurityException localSecurityException = null;
+                if (localException instanceof ElasticsearchSecurityException securityException) localSecurityException = securityException;
+
                 boolean foundFlat = false;
                 // checking if flat expression matched remotely
                 for (String remoteExpression : remoteExpressions) {
                     String[] splitResource = splitQualifiedResource(remoteExpression);
-                    ElasticsearchException exception = checkSingleRemoteExpression(
+                    var projectAlias = splitResource[0];
+                    var resource = splitResource[1];
+
+                    ElasticsearchException remoteException = checkSingleRemoteExpression(
                         remoteResolvedExpressions,
-                        splitResource[0], // projectAlias
-                        splitResource[1], // resource
+                        projectAlias,
+                        resource,
                         remoteExpression,
                         indicesOptions
                     );
-                    if (exception == null) {
+                    if (remoteException == null) {
                         // found flat expression somewhere
                         foundFlat = true;
+                        // ensure we don't report this index as unauthorized if it's unauthorized on another remote
+                        if (remoteUnauthorizedIndices != null) {
+                            for (var entry : remoteUnauthorizedIndices.entrySet()) {
+                                entry.setValue(entry.getValue().stream().filter(index -> index.endsWith(":" + resource) == false).toList());
+                            }
+                        }
                         break;
                     }
-                    if (unauthorizedException == null && exception instanceof ElasticsearchSecurityException securityException) {
-                        unauthorizedException = securityException;
+                    if (localSecurityException == null && remoteException instanceof ElasticsearchSecurityException securityException) {
+                        if (remoteAuthorizationExceptions == null) {
+                            remoteAuthorizationExceptions = new HashMap<>();
+                            remoteUnauthorizedIndices = new HashMap<>();
+                        }
+                        remoteAuthorizationExceptions.putIfAbsent(projectAlias, securityException);
+                        remoteUnauthorizedIndices.computeIfAbsent(projectAlias, k -> new ArrayList<>()).add(remoteExpression);
```

Comment:
I think this means, when local exception is not 403, we will report 403 from all linked projects where it is unauthorized. But this expression is unqualified and as discussed over the serverless PR, we'd report 403 for only the first project where it's unauthorized. That is, if the expression is unauthorized for local project, we report it for the local project only, otherwise we report it for the first linked project where it is unauthorized.

This is different from qualified expression, e.g. `*:logs`, for which we report 403 from all projects.